src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019-present HiveMQ and the HiveMQ Community
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hivemq.cli.utils.broker;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class TlsConfiguration {
+
+    private final boolean tlsEnabled;
+    private final @NotNull List<TlsVersion> tlsVersions;
+    private final @NotNull List<String> cipherSuites;
+    private final boolean clientAuthentication;
+
+    public TlsConfiguration(
+            final boolean tlsEnabled,
+            final @NotNull List<TlsVersion> tlsVersions,
+            final @NotNull List<String> cipherSuites,
+            final boolean clientAuthentication) {
+        this.tlsEnabled = tlsEnabled;
+        this.tlsVersions = tlsVersions;
+        this.cipherSuites = cipherSuites;
+        this.clientAuthentication = clientAuthentication;
+    }
+
+    public static Builder builder() {
+        return new TlsConfiguration.Builder();
+    }
+
+    public boolean isTlsEnabled() {
+        return tlsEnabled;
+    }
+
+    public @NotNull List<TlsVersion> getTlsVersions() {
+        return tlsVersions;
+    }
+
+    public @NotNull List<String> getCipherSuites() {
+        return cipherSuites;
+    }
+
+    public boolean isClientAuthentication() {
+        return clientAuthentication;
+    }
+
+
+    public static class Builder {
+
+        private boolean tlsEnabled = false;
+        private @NotNull List<TlsVersion> tlsVersions = List.of();
+        private @NotNull List<String> cipherSuites = List.of();
+        private boolean clientAuthentication = false;
+
+
+        public @NotNull Builder withTlsEnabled(final boolean tlsEnabled) {
+            this.tlsEnabled = tlsEnabled;
+            return this;
+        }
+
+        public @NotNull Builder withTlsVersions(final @NotNull List<TlsVersion> tlsVersions) {
```

Comment:
I would suggest a builder method that takes the `TlsVersion` enum, so you don't have to copy&paste the values in all usages. This would make it easier to test the commented out versions or to add future versions. This is a Git patch of my suggestion:
```
Index: src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java b/src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java
--- a/src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/utils/broker/TlsConfiguration.java	(date 1677505042845)
@@ -18,6 +18,7 @@
 
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Arrays;
 import java.util.List;
 
 public class TlsConfiguration {
@@ -58,7 +59,6 @@
         return clientAuthentication;
     }
 
-
     public static class Builder {
 
         private boolean tlsEnabled = false;
@@ -66,18 +66,20 @@
         private @NotNull List<String> cipherSuites = List.of();
         private boolean clientAuthentication = false;
 
-
         public @NotNull Builder withTlsEnabled(final boolean tlsEnabled) {
             this.tlsEnabled = tlsEnabled;
             return this;
         }
 
+        public @NotNull Builder withTlsVersions(final @NotNull Class<TlsVersion> tlsVersionClass) {
+            return withTlsVersions(Arrays.asList(tlsVersionClass.getEnumConstants()));
+        }
+
         public @NotNull Builder withTlsVersions(final @NotNull List<TlsVersion> tlsVersions) {
             this.tlsVersions = tlsVersions;
             return this;
         }
 
-
         public @NotNull Builder withCipherSuites(final @NotNull List<String> cipherSuites) {
             this.cipherSuites = cipherSuites;
             return this;
@@ -93,5 +95,3 @@
         }
     }
 }
-
-
Index: src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectTlsST.java	(date 1677502839611)
@@ -49,7 +49,7 @@
     @SuppressWarnings("JUnitMalformedDeclaration")
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .withCipherSuites(CipherSuite.getAllAsString())
                     .withTlsEnabled(true)
                     .build())
Index: src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectMTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectMTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectMTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectMTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/publish/PublishConnectMTlsST.java	(date 1677502839627)
@@ -43,7 +43,7 @@
     @SuppressWarnings("JUnitMalformedDeclaration")
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .withClientAuthentication(true)
                     .withTlsEnabled(true)
                     .build())
Index: src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectTlsST.java	(date 1677503154540)
@@ -41,7 +41,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .build())
             .build();
 
Index: src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectMTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectMTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectMTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectMTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectMTlsST.java	(date 1677503154556)
@@ -46,7 +46,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .withClientAuthentication(true)
                     .build())
             .build();
Index: src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectMTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectMTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectMTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectMTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/shell/connect/ShellConnectMTlsST.java	(date 1677503154504)
@@ -40,7 +40,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .build())
             .build();
 
Index: src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectTlsST.java	(date 1677503154564)
@@ -46,7 +46,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .build())
             .build();
 
Index: src/systemTest/java/com/hivemq/cli/commands/cli/CipherSuitesTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/CipherSuitesTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/CipherSuitesTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/CipherSuitesTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/CipherSuitesTlsST.java	(date 1677502839619)
@@ -43,7 +43,7 @@
     @SuppressWarnings("JUnitMalformedDeclaration")
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .withCipherSuites(CipherSuite.getAllAsString())
                     .withTlsEnabled(true)
                     .build())
Index: src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectMTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectMTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectMTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectMTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/test_broker/TestBrokerConnectMTlsST.java	(date 1677503154548)
@@ -41,7 +41,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .withClientAuthentication(true)
                     .build())
             .build();
Index: src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectTlsST.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectTlsST.java b/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectTlsST.java
--- a/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectTlsST.java	(revision 96b5d8b5bc4908b120ba502a34b84eaafa3e1f79)
+++ b/src/systemTest/java/com/hivemq/cli/commands/cli/subscribe/SubscribeConnectTlsST.java	(date 1677503154528)
@@ -51,7 +51,7 @@
     private final @NotNull HiveMQExtension HIVEMQ = HiveMQExtension.builder()
             .withTlsConfiguration(TlsConfiguration.builder()
                     .withTlsEnabled(true)
-                    .withTlsVersions(List.of(TlsVersion.TLS_1_2, TlsVersion.TLS_1_3))
+                    .withTlsVersions(TlsVersion.class)
                     .build())
             .build();
 
```

Reply:
I added a supportedAsList() method to the TlsVersion enum to allow similar simplification.